### PR TITLE
fix: selected video attachment in attachment picker isn't recognized as selected and upload issues on android

### LINKED ISSUE #1925: selected video attachment in attachment picker isnt recognized as selected
**Describe the bug**
In iOS Simulator and on iOS device, open attachment picker and select video. Video correctly show up in the message input, but does not receive the selected overlay UI. 

Seems to be related to https://github.com/GetStream/stream-chat-react-native/pull/1440/files; the file.uri and the asset.uri don't match because file uri is using the localAssetURI value, asset uri is using the original uri value, and the check for selected tries to match asset.uri to file.uri `selectedFiles.some((file) => file.uri === asset.uri`

Selected UI seems to work on android.

**Dev environment info (please complete/provide the following information):**

    "stream-chat": "^8.3.0",
    "stream-chat-expo": "^5.8.0",

**To Reproduce**

open attachment picker and select a video

**Expected behavior**

expected to see selected UI on a selected video in attachment picker

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

Fixes #1925

## 🛠 Implementation details

The URI is stored in the selected files and the final URI is extracted when uploading the image using `getLocalAssetUri`. The types are also fixed/simplified for the `doDocUploadRequest`.

## 🎨 UI Changes

- Expo iOS

https://user-images.githubusercontent.com/39884168/214798707-8a7c8363-c7f9-4200-9f93-8053998c4de2.mp4

- Expo Android

https://user-images.githubusercontent.com/39884168/214798613-c0eaa366-d63c-4b9f-b3eb-bdf3f7d71fbc.mov

- Native iOS

https://user-images.githubusercontent.com/39884168/214798334-6395514a-0846-4626-91e8-d755872f414f.mp4

- Native Android

https://user-images.githubusercontent.com/39884168/214799074-cd1f46c6-49b1-4a0a-9d93-1ba1826f82cc.mov

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

**Comment by vanGalilea** [comment]
Hey @khushal87 all works well besides no sound on expo ios when video is played

File: package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx [hunk 1]
```diff
@@ -64,9 +63,8 @@ const AttachmentVideo: React.FC<AttachmentVideoProps> = (props) => {
 
   const size = vw(100) / (numberOfAttachmentPickerImageColumns || 3) - 2;
 
-  const onPressVideo = async () => {
+  const onPressVideo = () => {
     // For the case of expo messaging app where you need to fetch the asset uri from asset id
```
Comment (vanGalilea) [inline package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx:67]: This line is not relevant anymore can we remove this comment?

File: package/src/contexts/messageInputContext/MessageInputContext.tsx [hunk 2]
```diff
@@ -991,23 +988,9 @@ export const MessageInputProvider = <
...
-        if (file.id) {
-          return await getLocalAssetUri(file.id);
-        } else if (file.uri?.match(/assets-library/)) {
-          return await getLocalAssetUri(file.uri);
-        }
-        return file.uri;
-      };
-      const uri = file.name || (await getLocalUri()) || '';
+      // For the case of expo messaging app where you need to fetch the file uri from file id. Here it is only done for iOS since for android the file.uri is fine.
+      const localAssetURI = Platform.OS === 'ios' && file.id && (await getLocalAssetUri(file.id));
```
Comment (vanGalilea) [inline package/src/contexts/messageInputContext/MessageInputContext.tsx:992]: Can we incorporate this logic in `getLocalAssetUri` function? because DRY

Comment (khushal87) [reply package/src/contexts/messageInputContext/MessageInputContext.tsx:992]: `getLocalAssetUri` is a native util. I don't think it makes sense to introduce such an logic inside it

Comment (vanGalilea) [reply package/src/contexts/messageInputContext/MessageInputContext.tsx:992]: 👍🏻

File: package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx (hunk 1 above)
Comment (vanGalilea) [inline package/src/components/AttachmentPicker/components/AttachmentPickerItem.tsx:67]:
```suggestion
```

**Pull request merged**